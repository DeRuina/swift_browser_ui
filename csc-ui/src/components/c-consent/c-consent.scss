:host {
  display: flex;
  align-items: center;
  position: fixed;
  z-index: 20000;
  bottom: 0;
  left: 0;
  right: 0;
  background: var(--csc-light-grey-blue);
  border-top: 1px var(--csc-primary) solid;
  color: var(--csc-dark-grey);
  min-height: 100px;
}

:host > c-row {
  flex: 1;
  margin: 24px;
}
