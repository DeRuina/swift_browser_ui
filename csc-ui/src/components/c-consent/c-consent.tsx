import { Component, Host, h } from '@stencil/core';
/**
 * A container component for cookie consent
 *
 * @group Popups
 * @slot - Cookie consent content
 */
@Component({
  tag: 'c-consent',
  styleUrl: 'c-consent.scss',
  shadow: true,
})
export class CConsent {
  render() {
    return (
      <Host>
        <c-row align="center" gap={8}>
          <slot></slot>
        </c-row>
      </Host>
    );
  }
}
