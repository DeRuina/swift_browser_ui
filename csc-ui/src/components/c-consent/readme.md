# c-consent



<!-- Auto Generated Below -->


## Overview

A container component for cookie consent

## Slots

| Slot | Description            |
| ---- | ---------------------- |
|      | Cookie consent content |


## Dependencies

### Depends on

- [c-row](../c-row)

### Graph
```mermaid
graph TD;
  c-consent --> c-row
  style c-consent fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
