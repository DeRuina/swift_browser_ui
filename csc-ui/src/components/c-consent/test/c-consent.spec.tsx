import { newSpecPage } from '@stencil/core/testing';
import { CConsent } from '../c-consent';

describe('c-consent', () => {
  it('renders default html', async () => {
    const page = await newSpecPage({
      components: [CConsent],
      html: `<c-consent>
      <span>Show your consent</span>
      </c-consent>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
