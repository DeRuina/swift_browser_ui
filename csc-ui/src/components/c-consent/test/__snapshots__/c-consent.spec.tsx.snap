// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-consent renders default html 1`] = `
<c-consent>
  <mock:shadow-root>
    <c-row align="center" gap="8">
      <slot></slot>
    </c-row>
  </mock:shadow-root>
  <span>
    Show your consent
  </span>
</c-consent>
`;
